src/placeholder/solvers/interface.py

```diff
@@ -0,0 +1,92 @@
+
+class ModelNotDefinedError(Exception):
+    pass
+
+
+class SolverNotDefinedError(Exception):
+    pass
+
+
+class Solver:
+
+    def __init__(self, model_instance=None):
+        self._model = model_instance
+        self.success = None
+        self.x_opt = None
+        self.fun_val_opt = None
+        self.success = None
+
+    @property
+    def model(self):
+        return self._model
+
+    @model.setter
+    def model(self, model_instance):
+        self._model = model_instance
+
+    def assert_models_defined(self):
+        if self._model is not None:
+            return True
+        else:
+            raise ModelNotDefinedError()
+
+    def fit(self, data, x_init=None, options=None):
+        raise NotImplementedError()
+
+    def predict(self, **kwargs):
+        raise NotImplementedError()
+
+
+class CompositeSolver(Solver):
+
+    def __init__(self, solvers_list=None):
+        super().__init__(model_instance=None)
+        if solvers_list is not None:
+            self._solvers = solvers_list
```

Comment:
How do we `fit` a `CompositeSolver` since it has multiple solvers?

Reply:
I think it would depend on the concrete implementations. For instance, if we ensemble different solvers we would run all and take the best, or in gaussian mixture thing we would fit two sequentially, or in EM we may alternate between two (?), or in some other cases we may actually want to restrict the solvers to be only one. We do not have to have a list though, if we feel it may not be particularly useful or too complicated for our interface :)

Reply:
oops I meant just how do we call fit haha does `CompositeSolver` have a `fit` method?

Reply:
ah right, I think it inherits the signature from `Solver`? (neither is implemented in the interface). sorry I think I accidentally added `predict` but not `fit` in the composite interface, which probably have caused the confusion...